core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -405,7 +405,8 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
 
             // Keep track of the number of unfinished requests and
             // clean up the request stream when response stream ends.
-            final boolean isTransient = service.as(TransientService.class) != null;
+            final boolean isTransient = service.as(TransientService.class) != null ||
+                                        serviceCfg.transientService();
```

Comment:
We should only check `ServiceConfig.transientService()` and set `true` to it when `service.as(TransientService.class) != null` *and* a user did not override it with `transientService(false)`.